File: tella-vault/src/main/java/com/hzontal/tella_vault/VaultFile.java
package com.hzontal.tella_vault;

import java.io.Serializable;

public class VaultFile  implements Serializable {
    public String id; // generated uuid
    public Type type;
    public String hash;
    public String path;
    public String mimeType = null;
    public String name;
    public long size;
    public long created;
    public long duration;
    public boolean anonymous;
    public Metadata metadata;
    public byte[] thumb = null;

    public VaultFile() {
    }

    protected VaultFile(BaseVaultFileBuilder<?, ?> builder) {
        this.id = builder.id;
        this.name = builder.name;
        this.type = builder.type;
        this.mimeType = builder.mimeType;
        this.thumb = builder.thumb;
        this.metadata = builder.metadata;
        this.anonymous = builder.anonymous;
        this.path = builder.path;
        this.hash = builder.hash;
        this.duration = builder.duration;
        this.size = builder.size;
    }

    public enum Type {
        UNKNOWN(0),
        FILE(1),
        DIRECTORY(2);

        private final int value;

        Type(int value) {
            this.value = value;
        }

        public int getValue() {
            return value;
        }

        public static Type fromValue(int value) {
            for (Type v: Type.values()) {
                if (v.getValue() == value) return v;
            }

            return UNKNOWN;
        }
    }

    @Override
    public boolean equals(Object o) {

        try {
            if (o == null || getClass() != o.getClass())
                return false;

            VaultFile p = (VaultFile) o;

            return ((id.equals(p.id))
                    && (name.equals(p.name))
                    && (type.equals(p.type))
                    && (mimeType.equals(p.mimeType))
                    && (hash.equals(p.hash))
                    && (duration == p.duration)
                    && (size == p.size));
        } catch (Exception ex) {
            return false;
        }
    }
}

File: mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/vault/adapters/viewholders/data/VaultFile.kt
package rs.readahead.washington.mobile.views.fragment.vault.adapters.viewholders.data

import rs.readahead.washington.mobile.domain.entity.MediaFile

data class VaultFile (var id // generated uuid
                      : Int,
                      var type: MediaFile.Type? = null,
                      var hash: String? = null,
                      var path: String? = null,
                      var mimeType: String? = null,
                      var name: String = "",
                      var size: Long = 0,
                      var created: Long = 0,
                      var duration: Long = 0,
                      var anonymous : Boolean = false,
                      var metadata: Metadata? = null,
                      var thumb: ByteArray?) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as VaultFile

        if (id != other.id) return false
        if (type != other.type) return false
        if (hash != other.hash) return false
        if (path != other.path) return false
        if (mimeType != other.mimeType) return false
        if (name != other.name) return false
        if (size != other.size) return false
        if (created != other.created) return false
        if (duration != other.duration) return false
        if (anonymous != other.anonymous) return false
        if (metadata != other.metadata) return false
        if (!other.thumb?.let { thumb?.contentEquals(it) }!!) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + (type?.hashCode() ?: 0)
        result = 31 * result + (hash?.hashCode() ?: 0)
        result = 31 * result + (path?.hashCode() ?: 0)
        result = 31 * result + (mimeType?.hashCode() ?: 0)
        result = 31 * result + (name.hashCode() ?: 0)
        result = 31 * result + size.hashCode()
        result = 31 * result + created.hashCode()
        result = 31 * result + duration.hashCode()
        result = 31 * result + anonymous.hashCode()
        result = 31 * result + (metadata?.hashCode() ?: 0)
        result = 31 * result + thumb?.contentHashCode()!!
        return result
    }
}


